databricks/koalas/indexes/base.py

```diff
@@ -353,8 +353,10 @@ def equals(self, other) -> bool:
                     # some exceptions when 'compute.ops_on_diff_frames' is enabled.
                     # Working around for now via using frame.
                     return (
-                        self.to_series("self").reset_index(drop=True)
-                        == other.to_series("other").reset_index(drop=True)
+                        IndexOpsMixin.__eq__(
+                            self.to_series("self").reset_index(drop=True),
+                            other.to_series("other").reset_index(drop=True),
+                        )
```

Comment:
Series comparison with `==` operator uses `Index.equals` internally.

So it falls eternal recursive if we use `==` for Series comparison here.